### PR TITLE
fix(actionlib): Remove `getState` error when no goal is running

### LINKED ISSUE #40: SimpleActionClient is tracking goal
I have reached a point in my application in which I would like to have a SimpleActionClient member which allows me to know if the server is currently pursuing a goal. Something like `action_server.isTrackingGoal()`.

I guess is possible (I haven't checked in detail the SimpleActionClient implementation). Is there any reason not to do this? I think I could try out an implementation in case it is possible.

### PR DESCRIPTION
**Pull request by alireza-hosseini**
As explained in #40 in many situations the developer needs to know if the action client is tracking a goal or not which is possible by the `getState` method of the `SimpleActionClient`. But the problem is, using this function while the action client didn't send any goal before, results to the following error:

> Trying to getState() when no goal is running. You are incorrectly using SimpleActionClient

This PR simply removes this ROS error since it's not really an incorrect usage of the `SimpleActionClient` (closes #40).

### CODE REVIEW

**Comment by alireza-hosseini** [comment]
I checked the Jenkins report for the failed jobs, apparently, they are not related to my changes.

**Comment by alireza-hosseini** [comment]
Is there any plan to review this simple PR?

**Comment by jschleicher** [comment]
Looks good to me. @mikaelarguedas Could you please review and merge this change?

**Comment by mikaelarguedas** [comment]
@alireza-hosseini @jschleicher This looks good to me, I agree it's not really an invalid usage of the simple action client.
I originally held this as I was planning on releasing it only for ROS Melodic to not change behavior for people that may be expecting these logging messages.

I'll leave it up to the current maintainer to review / merge / release but you have my :+1:

**Comment by jschleicher** [comment]
@mikaelarguedas merging this for kinetic onwards would be fine for me. On the other hand this logging message shouldn't break any existing code (and is ABI compatible of course).

Who is the current maintainer?

**Comment by mikaelarguedas** [comment]
> On the other hand this logging message shouldn't break any existing code (and is ABI compatible of course).

agreed :+1:

> Who is the current maintainer?

@mjcarroll is the current maintainer

**Review by mjcarroll — APPROVED** [review]
LGTM, thanks for the contribution.

**Pull request merged**
